monix-tail/shared/src/main/scala/monix/tail/batches/Batch.scala

```diff
@@ -163,19 +163,19 @@ object Batch {
     * @param offset $paramArrayOffset
     * @param length $paramArrayLength
     */
-  def fromArray[A : ClassTag](array: Array[A], offset: Int, length: Int): ArrayBatch[A] =
-    new ArrayBatch[A](array, offset, length)
+  def fromArray[A](array: Array[A], offset: Int, length: Int): ArrayBatch[A] = {
+    val tp = ClassTag[A](array.getClass.getComponentType)
+    new ArrayBatch[A](array, offset, length)(tp)
+  }
 
   /** $fromAnyArrayDesc
     *
     * @param array $paramArray
     * @param offset $paramArrayOffset
     * @param length $paramArrayLength
     */
-  def fromAnyArray[A](array: Array[_], offset: Int, length: Int): ArrayBatch[A] = {
-    val ref = new ArrayBatch[Any](array.asInstanceOf[Array[Any]], offset, length, arrayAnyBuilder)
```

Comment:
We can now remove `arrayAnyBuilder` in `monix.tail.batches` package object, right?

Reply:
Yes. Will do.

Reply:
I removed the package object completely. Thanks for pointing it out.